src/Microsoft.VisualStudio.Threading.Analyzers/VSSDK014AvoidLegacyThreadSwitchingAnalyzer.cs

```diff
@@ -0,0 +1,61 @@
+﻿namespace Microsoft.VisualStudio.Threading.Analyzers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Text;
+    using System.Threading.Tasks;
+    using CodeAnalysis;
+    using CodeAnalysis.CSharp;
+    using CodeAnalysis.CSharp.Syntax;
+    using CodeAnalysis.Diagnostics;
+
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class VSSDK014AvoidLegacyThreadSwitchingAnalyzer : DiagnosticAnalyzer
+    {
+        public const string Id = "VSSDK014";
+
+        internal static readonly DiagnosticDescriptor Descriptor = new DiagnosticDescriptor(
+            id: Id,
+            title: Strings.VSSDK014_Title,
+            messageFormat: Strings.VSSDK014_MessageFormat,
+            category: "Usage",
+            defaultSeverity: DiagnosticSeverity.Warning,
+            isEnabledByDefault: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
+
+        public override void Initialize(AnalysisContext context)
```

Comment:
Won't fix: Roslyn invokes this method and always supplies a value as part of its contract.
That, and we don't check it in any of the other analyzers.